app/keepers/types.go

```diff
@@ -0,0 +1,79 @@
+	icacontrollerkeeper "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/controller/keeper"
+	icahostkeeper "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/host/keeper"
+	ibcfeekeeper "github.com/cosmos/ibc-go/v8/modules/apps/29-fee/keeper"
+	ibctransferkeeper "github.com/cosmos/ibc-go/v8/modules/apps/transfer/keeper"
+	ibckeeper "github.com/cosmos/ibc-go/v8/modules/core/keeper"
+
+	applicationmodulekeeper "github.com/pokt-network/poktroll/x/application/keeper"
+	gatewaymodulekeeper "github.com/pokt-network/poktroll/x/gateway/keeper"
+	proofmodulekeeper "github.com/pokt-network/poktroll/x/proof/keeper"
+	servicemodulekeeper "github.com/pokt-network/poktroll/x/service/keeper"
+	sessionmodulekeeper "github.com/pokt-network/poktroll/x/session/keeper"
+	sharedmodulekeeper "github.com/pokt-network/poktroll/x/shared/keeper"
+	suppliermodulekeeper "github.com/pokt-network/poktroll/x/supplier/keeper"
+	tokenomicsmodulekeeper "github.com/pokt-network/poktroll/x/tokenomics/keeper"
+)
+
+// Keepers includes all possible keepers. We separated it into a separate struct to make it easier to scaffold upgrades.
+type Keepers struct {
+	AccountKeeper         authkeeper.AccountKeeper
+	BankKeeper            bankkeeper.Keeper
+	StakingKeeper         *stakingkeeper.Keeper
+	SlashingKeeper        slashingkeeper.Keeper
+	MintKeeper            mintkeeper.Keeper
+	DistrKeeper           distrkeeper.Keeper
+	GovKeeper             *govkeeper.Keeper
+	CrisisKeeper          *crisiskeeper.Keeper
+	UpgradeKeeper         *upgradekeeper.Keeper
+	ParamsKeeper          paramskeeper.Keeper
+	AuthzKeeper           authzkeeper.Keeper
+	EvidenceKeeper        evidencekeeper.Keeper
+	FeeGrantKeeper        feegrantkeeper.Keeper
+	GroupKeeper           groupkeeper.Keeper
+	ConsensusParamsKeeper consensuskeeper.Keeper
+	CircuitBreakerKeeper  circuitkeeper.Keeper
+
+	// IBC
+	IBCKeeper           *ibckeeper.Keeper // IBC Keeper must be a pointer in the app, so we can SetRouter on it correctly
+	CapabilityKeeper    *capabilitykeeper.Keeper
+	IBCFeeKeeper        ibcfeekeeper.Keeper
+	ICAControllerKeeper icacontrollerkeeper.Keeper
+	ICAHostKeeper       icahostkeeper.Keeper
+	TransferKeeper      ibctransferkeeper.Keeper
+
+	// Scoped IBC
+	ScopedIBCKeeper           capabilitykeeper.ScopedKeeper
+	ScopedIBCTransferKeeper   capabilitykeeper.ScopedKeeper
+	ScopedICAControllerKeeper capabilitykeeper.ScopedKeeper
+	ScopedICAHostKeeper       capabilitykeeper.ScopedKeeper
+
+	ServiceKeeper     servicemodulekeeper.Keeper
+	GatewayKeeper     gatewaymodulekeeper.Keeper
+	ApplicationKeeper applicationmodulekeeper.Keeper
+	SupplierKeeper    suppliermodulekeeper.Keeper
+	SessionKeeper     sessionmodulekeeper.Keeper
+	ProofKeeper       proofmodulekeeper.Keeper
+	TokenomicsKeeper  tokenomicsmodulekeeper.Keeper
+	SharedKeeper      sharedmodulekeeper.Keeper
+	// this line is used by starport scaffolding # stargate/app/keeperDeclaration
```

Comment:
1. I had to do this to avoid a cyclic dependency between `upgrade` and `app` packages.
2. I just checked - the scaffolding fails for me for other reasons (we import other packages differently), but looking at cosmos sdk code, this might not actually work. Let's move this line back to `app.go` and add a comment "move to `keepers` package after scaffolding".

Reply:
1. I had to do this to avoid a cyclic dependency between `upgrade` and `app` packages. This structure allows to pass all available keepers into the upgrade function:
```go
CreateUpgradeHandler func(*module.Manager, *keepers.Keepers, module.Configurator) upgradetypes.UpgradeHandler
```
3. I just checked - the scaffolding fails for me for other reasons (we import other packages differently), but looking at ignite code, this might not actually work. Let's move this line back to `app.go` and add a comment "move to `keepers` package after scaffolding".